docs/enhancements/1-crds-and-controllers.md

```diff
@@ -0,0 +1,430 @@
+| Status | Authors   | Created    | 
+|--------|-----------|------------|
+| Review | @jewertow | Jan 2 2025 |
+
+# CRDs and controllers
+
+## Overview
+
+Currently, we use helm to manage the controller deployment and its configuration. The configuration is passed to the controller as CLI arguments.
+This approach requires restarting the controller on any configuration change and causes pod failures in case of a misconfiguration.
+
+Additionally, current approach to resource management is not reliable. We apply resources when it's necessary using the Istio client's `Apply` function,
+so we are not able to reconcile created objects on user's modifications or controller uninstallation.
```

Comment:
```suggestion
We create required Istio resources only when necessary, driven exclusively by notifications from the FDS client. However, this approach has a drawback - it is not reliable and does not ensure the desired state of the cluster after resources are created. Since there is no ongoing reconciliation, any drift from the intended configuration, whether due to user modifications or controller uninstallation, remains uncorrected.
```